clang complaining on OSX 10.11.5

Tried to compile against tip go gbm-developers/gbm 
commit 8eca6787194745b68a2310e68341b7eb34aa7db4

R version 3.2.3 (2015-12-10) -- "Wooden Christmas-Tree"
Copyright (C) 2015 The R Foundation for Statistical Computing
Platform: x86_64-apple-darwin14.5.0 (64-bit)

R is free software and comes with ABSOLUTELY NO WARRANTY.
You are welcome to redistribute it under certain conditions.
Type 'license()' or 'licence()' for distribution details.

  Natural language support but running in an English locale

R is a collaborative project with many contributors.
Type 'contributors()' for more information and
'citation()' on how to cite R or R packages in publications.

Type 'demo()' for some demos, 'help()' for on-line help, or
'help.start()' for an HTML browser interface to help.
Type 'q()' to quit R.

Microsoft R Open 3.2.3
Default CRAN mirror snapshot taken on 2016-01-01
The enhanced R distribution from Microsoft
Visit http://go.microsoft.com/fwlink/?LinkID=722555 for information
about additional features.

Multithreaded BLAS/LAPACK libraries detected. Using 4 cores for math algorithms.
> devtools::check()
Updating gbm documentation
Loading gbm
Re-compiling gbm
'/Library/Frameworks/R.framework/Resources/bin/R' --no-site-file --no-environ  \
  --no-save --no-restore CMD INSTALL '/Users/druau/Box Sync/Project -  \
  Structured Exploration/code/gbm'  \
  --library='/var/folders/cr/x4x8pv151zqgktflqn3sz3cnl3mln2/T//Rtmpj0bQum/devtools_install_ae4a2ea264ef'  \
  --no-R --no-data --no-help --no-demo --no-inst --no-docs --no-exec  \
  --no-multiarch --no-test-load --preclean 

* installing *source* package ‘gbm’ ...
** libs

--- [blablabla...] ---

clang++ -I/Library/Frameworks/R.framework/Resources/include -DNDEBUG  -I/usr/local/include -I/usr/local/include/freetype2 -I/opt/X11/include -DPLATFORM_PKGTYPE='"mac.binary.mavericks"' -I"/Library/Frameworks/R.framework/Versions/3.2/Resources/library/Rcpp/include"   -fPIC  -Wall -mtune=core2 -g -O2  -c gamma.cpp -o gamma.o
gamma.cpp:35:45: warning: reference cannot be bound to dereferenced null pointer in well-defined C++ code; pointer may be assumed to always convert to true [-Wundefined-bool-conversion]
  if (!(kData.y_ptr() && kFuncEstimate && &(residuals[0]) &&
                                       ~~  ~^~~~~~~~~~~~~

clang++ -I/Library/Frameworks/R.framework/Resources/include -DNDEBUG  -I/usr/local/include -I/usr/local/include/freetype2 -I/opt/X11/include -DPLATFORM_PKGTYPE='"mac.binary.mavericks"' -I"/Library/Frameworks/R.framework/Versions/3.2/Resources/library/Rcpp/include"   -fPIC  -Wall -mtune=core2 -g -O2  -c gaussian.cpp -o gaussian.o
gaussian.cpp:32:45: warning: reference cannot be bound to dereferenced null pointer in well-defined C++ code; pointer may be assumed to always convert to true [-Wundefined-bool-conversion]
  if (!(kData.y_ptr() && kFuncEstimate && &(residuals[0]) &&
                                       ~~  ~^~~~~~~~~~~~~

clang++ -I/Library/Frameworks/R.framework/Resources/include -DNDEBUG  -I/usr/local/include -I/usr/local/include/freetype2 -I/opt/X11/include -DPLATFORM_PKGTYPE='"mac.binary.mavericks"' -I"/Library/Frameworks/R.framework/Versions/3.2/Resources/library/Rcpp/include"   -fPIC  -Wall -mtune=core2 -g -O2  -c tdist.cpp -o tdist.o
In file included from tdist.cpp:13:
In file included from ./tdist.h:17:
In file included from ./distribution.h:24:
In file included from ./dataset.h:21:
In file included from ./datadistparams.h:22:
In file included from /Library/Frameworks/R.framework/Versions/3.2/Resources/library/Rcpp/include/Rcpp.h:27:
In file included from /Library/Frameworks/R.framework/Versions/3.2/Resources/library/Rcpp/include/RcppCommon.h:122:
/Library/Frameworks/R.framework/Versions/3.2/Resources/library/Rcpp/include/Rcpp/exceptions.h:196:53: warning: all paths through this function will call itself [-Winfinite-recursion]
    inline void warning(const std::string& message) {
                                                    ^

clang++ -I/Library/Frameworks/R.framework/Resources/include -DNDEBUG  -I/usr/local/include -I/usr/local/include/freetype2 -I/opt/X11/include -DPLATFORM_PKGTYPE='"mac.binary.mavericks"' -I"/Library/Frameworks/R.framework/Versions/3.2/Resources/library/Rcpp/include"   -fPIC  -Wall -mtune=core2 -g -O2  -c tweedie.cpp -o tweedie.o
tweedie.cpp:48:46: warning: reference cannot be bound to dereferenced null pointer in well-defined C++ code; pointer may be assumed to always convert to true [-Wundefined-bool-conversion]
  if (!(kData.y_ptr() && kFuncEstimates && &(residuals[0]) &&
                                        ~~  ~^~~~~~~~~~~~~

clang++ -dynamiclib -Wl,-headerpad_max_install_names -undefined dynamic_lookup -single_module -multiply_defined suppress -L/Library/Frameworks/R.framework/Resources/lib -L/opt/X11/lib -L/usr/local/lib /usr/local/lib/libcairo.a /usr/local/lib/libpixman-1.a /usr/local/lib/libfreetype.a /usr/local/lib/libfontconfig.a -lxml2 /usr/local/lib/libreadline.a -o gbm.so adaboost.o bernoulli.o coxph.o dataset.o distribution.o distribution_factory.o gamma.o gaussian.o gbm_datadistcontainer.o gbm_engine.o gbm_fit.o gbm_functions.o gbmentry.o huberized.o laplace.o locationm.o node.o node_parameters.o node_search.o pairwise.o poisson.o quantile.o tdist.o tree.o tweedie.o varsplitter.o -F/Library/Frameworks/R.framework/.. -framework R -Wl,-framework -Wl,CoreFoundation
clang: error: no such file or directory: '/usr/local/lib/libreadline.a'
make: *** [gbm.so] Error 1
ERROR: compilation failed for package ‘gbm’
* removing ‘/private/var/folders/cr/x4x8pv151zqgktflqn3sz3cnl3mln2/T/Rtmpj0bQum/devtools_install_ae4a2ea264ef/gbm’
Error: Command failed (1)

--- a/src/gamma.cpp
+++ b/src/gamma.cpp
@@ -32,7 +32,7 @@ CGamma::~CGamma() {}
 void CGamma::ComputeWorkingResponse(const CDataset& kData, const Bag& kBag,
                                     const double* kFuncEstimate,
                                     std::vector<double>& residuals) {
-  if (!(kData.y_ptr() && kFuncEstimate && &(residuals[0]) &&
+  if (!(kData.y_ptr() && kFuncEstimate &&
         kData.weight_ptr())) {
     throw gbm_exception::InvalidArgument();
   }

--- a/src/gaussian.cpp
+++ b/src/gaussian.cpp
@@ -29,7 +29,7 @@ CGaussian::~CGaussian() {}
 void CGaussian::ComputeWorkingResponse(const CDataset& kData, const Bag& kBag,
                                        const double* kFuncEstimate,
                                        std::vector<double>& residuals) {
-  if (!(kData.y_ptr() && kFuncEstimate && &(residuals[0]) &&
+  if (!(kData.y_ptr() && kFuncEstimate &&
         kData.weight_ptr())) {
     throw gbm_exception::InvalidArgument();
   }

--- a/src/tweedie.cpp
+++ b/src/tweedie.cpp
@@ -45,7 +45,7 @@ void CTweedie::ComputeWorkingResponse(const CDataset& kData, const Bag& kBag,
   unsigned long i = 0;
   double delta_func_est = 0.0;
 
-  if (!(kData.y_ptr() && kFuncEstimates && &(residuals[0]) &&
+  if (!(kData.y_ptr() && kFuncEstimates &&
         kData.weight_ptr())) {
     throw gbm_exception::InvalidArgument();
   }

